sunpy/util/scraper.py

```diff
@@ -116,8 +116,8 @@ def range(self, timerange):
             Notice that these directories may not exist in the archive.
         """
         # find directory structure - without file names
-        directorypattern = os.path.dirname(self.pattern) + '/'
-        # TODO what if there's not slashes?
+        if '/' in self.pattern:
+            directorypattern = '/'.join(self.pattern.split('/')[:-1]) + '/'
```

Comment:
I don't understand how this fixed it on windows.

Reply:
So `os.path.dirname` calls different code paths on windows and nix/posix systems. On windows `\` is obviously a path separator [1](https://github.com/python/cpython/blob/216cb1469f566ba5493bf53a73da9ccdac05ccfc/Lib/ntpath.py#L34)  while on posix it's `/`  [2](https://github.com/python/cpython/blob/216cb1469f566ba5493bf53a73da9ccdac05ccfc/Lib/posixpath.py#L41) same as urls. So the `\` in the regex is treated as a path separator on windows.